mule-extensions-api/src/main/java/org/mule/runtime/extension/privileged/util/ComponentDeclarationUtils.java

```diff
@@ -0,0 +1,103 @@
+import org.mule.runtime.api.meta.model.declaration.fluent.ComponentDeclarer;
+import org.mule.runtime.api.meta.model.declaration.fluent.ExtensionDeclarer;
+import org.mule.runtime.api.meta.model.declaration.fluent.OperationDeclarer;
+import org.mule.runtime.api.meta.model.declaration.fluent.ParameterDeclarer;
+import org.mule.runtime.api.meta.model.operation.OperationModel;
+import org.mule.runtime.api.meta.model.parameter.ParameterModel;
+import org.mule.runtime.extension.internal.property.NoConnectivityErrorModelProperty;
+import org.mule.runtime.extension.internal.property.NoErrorMappingModelProperty;
+import org.mule.runtime.extension.internal.property.NoReconnectionStrategyModelProperty;
+import org.mule.runtime.extension.internal.property.NoStreamingConfigurationModelProperty;
+import org.mule.runtime.extension.internal.property.NoTransactionalActionModelProperty;
+import org.mule.runtime.extension.internal.property.PagedOperationModelProperty;
+import org.mule.runtime.extension.internal.property.TargetModelProperty;
+import org.mule.runtime.extension.internal.property.TransactionalActionModelProperty;
+import org.mule.runtime.extension.internal.property.TransactionalTypeModelProperty;
+
+import java.util.Set;
+
+/**
+ * Provides a way to access the functionality on internal model properties when declaring components on crafted extensions.
+ * <p>
+ * Being {@code privileged]}, this is not intended to be used outside of the scope of crafted extension declarations.
+ * 
+ * @since 1.8
+ */
+public final class ComponentDeclarationUtils {
+
+  private ComponentDeclarationUtils() {
+    // nothing to do
+  }
+
+  public static final void withNoStreamingConfiguration(OperationDeclarer declarer) {
+    declarer.withModelProperty(new NoStreamingConfigurationModelProperty());
+  }
+
+  public static final void withNoTransactionalAction(OperationDeclarer declarer) {
+    declarer.withModelProperty(new NoTransactionalActionModelProperty());
+  }
+
+  public static final void withNoReconnectionStrategy(OperationDeclarer declarer) {
+    declarer.withModelProperty(new NoReconnectionStrategyModelProperty());
+  }
+
+  // account for use cases where this is declared at the extension level for all its operations
+  public static final void withNoReconnectionStrategy(ExtensionDeclarer declarer) {
+    declarer.withModelProperty(new NoReconnectionStrategyModelProperty());
+  }
+
+  public static final void withNoConnectivityError(OperationDeclarer declarer) {
+    declarer.withModelProperty(new NoConnectivityErrorModelProperty());
+  }
+
+  public static final void withNoErrorMapping(OperationDeclarer declarer) {
+    declarer.withModelProperty(new NoErrorMappingModelProperty());
+  }
+
+  public static final void pagedOperation(ComponentDeclarer declarer) {
+    declarer.withModelProperty(new PagedOperationModelProperty());
+  }
+
+  public static final void targetModelProperty(ParameterDeclarer declarer) {
```

Comment:
this is only used to manually declare target parameters in runtime models, in (hopefully) the exact same way as `org.mule.runtime.extension.internal.loader.enricher.TargetParameterDeclarationEnricher.WalkDelegateDelegateDeclaration#enrichWithTargetValueParameterDeclaration`. 

Why do we need that? why doesn't the enricher kicks in and adds that parameter automatically? If there's a good reason for that, then this method should be called `addTargetParameter`, receive a complete ComponentDeclarer and execute the same code the enricher has today (the enricher should probably delegate into this one).